### ISSUE TITLE
GitVersion 5.1.2 on MacOS fails with "The application to execute does not exist"

### ISSUE THREAD

**Issue by tegataiprime** [GitTools/GitVersion#1920]
Installed via homebrew, but executable not found :(

```
$ uname -a
Darwin 18.7.0 Darwin Kernel Version 18.7.0: Sat Oct 12 00:02:19 PDT 2019; root:xnu-4903.278.12~1/RELEASE_X86_64 x86_64

$ brew install gitversion
==> Downloading https://github.com/GitTools/GitVersion/releases/download/5.1.2/gitversion-osx-5.1.2.tar.gz
==> Downloading from https://github-production-release-asset-2e65be.s3.amazonaws.com/12399790/f0354800-019
######################################################################## 100.0%
🍺  /usr/local/Cellar/gitversion/5.1.2: 4 files, 43.4MB, built in 5 seconds

$ gitversion --version
The application to execute does not exist: '/private/var/folders/5t/6h4jfhds54b1drrrrfybg5c80000gn/T/.net/GitVersion/q24if2il.0k5/GitVersion.dll'
```

**Comment by phoenixadb** [comment]
Same issue for me with the gitversion command :

`$ gitversion --version
The application to execute does not exist: '/private/var/folders/dd/z_59flyj1097lth482drv3j00000gp/T/.net/GitVersion/q24if2il.0k5/GitVersion.dll'.
$ uname -a
Darwin faas-mac-4 18.7.0 Darwin Kernel Version 18.7.0: Tue Aug 20 16:57:14 PDT 2019; root:xnu-4903.271.2~2/RELEASE_X86_64 x86_64`

**Comment by deeczhou** [comment]
ran into the same error: 
`$ gitversion
The application to execute does not exist: '/private/var/folders/kh/40zkn07n2mn0n5rx_6wk2d3w0000gn/T/.net/GitVersion/q24if2il.0k5/GitVersion.dll'.`

**Comment by mootalk** [comment]
Same here.
``` bash
$ gitversion
The application to execute does not exist: '/private/var/folders/q_/x3fg08t52dv52nzmjc7q98j80000gp/T/.net/GitVersion/ykbshukf.nm1/GitVersion.dll'.
```

**Comment by asbjornu** [comment]
Seems like we don't have a MacOS build. I seem to remember us having that at some point, but not anymore. We should probably get that up and running so we can detect problems like these. I wonder if this is only a problem with Homebrew, though. Does the global dotnet tool work?

**Comment by asbjornu** [comment]
@arturcic, can you remember if we had a macOS build going with Travis CI before? Could we get that up and running again and would that have been able to detect the problem reported here?

**Comment by arturcic** [comment]
@asbjornu well we do have build and unit test steps running on macos with Azure Pipeline. I'm planning to add support for macos and linux build on appveyor and later GitHub actions. Right now we test only the msbuild and global tool in linux containers. Was planning to add additional jobs for testing the single binary on all 3 platforms including macos. The single binary version is the single-file, self-contained version of the global tool, basically only one file specific for each platform.

**Comment by franknarf8** [comment]
I can also reproduce this error using the brew package for 5.1.3; is there a way to hotfix this?

**Comment by chb0github** [comment]
It was all working fine for me up until Today (last week it was good, no updates). Today I got
```
$ gitversion
The application to execute does not exist: '/private/var/folders/c4/m28c23954qq1ndnvlkv07rmshkh91h/T/.net/GitVersion/ykbshukf.nm1/GitVersion.dll'.

```
So I did 

```
brew uninstall gitversion
brew install gitversion
```

Same deal

**Comment by flin-8** [comment]
> It was all working fine for me up until Today (last week it was good, no updates). Today I got
> 
> ```
> $ gitversion
> The application to execute does not exist: '/private/var/folders/c4/m28c23954qq1ndnvlkv07rmshkh91h/T/.net/GitVersion/ykbshukf.nm1/GitVersion.dll'.
> ```
> 
> So I did
> 
> ```
> brew uninstall gitversion
> brew install gitversion
> ```
> 
> Same deal

We've seen an issue with a different dotnet executable (OctopusCLI) where some of the files in the tmp folder are deleted after a few days. 

Check to see if you are missing a few of those files.

Deleting the folder (`ykbshukf.nm1` in your case) should make it re-extract the files.

**Comment by flin-8** [comment]
Found some documentation [here](https://github.com/dotnet/designs/blob/master/accepted/2020/single-file/design_3_0.md#extraction-location):

If you set the `DOTNET_BUNDLE_EXTRACT_BASE_DIR` environment variable then you can change where the files are extracted.

**Comment by arturcic** [comment]
@flin-8 could you add to Gitversion docs the link you provided?

**Comment by flin-8** [comment]
> @flin-8 could you add to Gitversion docs the link you provided?

@artur-stolear-technosoft I'm not a regular contributor, so I wouldn't know where to start

On another note it looks like this bug has been fixed in [Netcore 3.1](https://github.com/dotnet/core-setup/commit/2364c9f773d5604e7fc5d0a1555c4b077e898583)!
